src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,216 @@
+        }
+        else if (args.length >= 5)
+        {
+            colonyId = TOO_MANY_ARGUMENTS_CODE;
+        }
+        else
+        {
+            colonyId = UNKNOWN_ERROR_CODE;
+        }
+        if (colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId) == null)
+            {
+                colonyId = NOT_FOUND_CODE;
+            }
+        }
+
+
+        return colonyId;
+    }
+
+    /**
+     * Getting the citizen ID.
+     * @param colonyId
+     * @param args
+     * @return
+     */
+    public static int getCitizenId(@NotNull final int colonyId, @NotNull final String... args)
+    {
+        int citizenId;
+        String citizenName;
+        citizenId = NOT_FOUND_CODE;
+        if (args.length == 2)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[1]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == 1)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == 3 && colonyId >= 0)
+        {
+            citizenName = args[0] + " " + args[1] + " " + args[2];
+            for (int i = 1
+                   ; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                if (ColonyManager.getColony(colonyId).getCitizen(i).getName() != null)
+                {
+                    if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                    {
+                        citizenId = i;
+                    }
+                }
+            }
+
+        }
+        else if (args.length == 4 && colonyId >= 0)
+        {
+            citizenName = args[1] + " " + args[2] + " " + args[3];
+            for (int i = 1; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                {
+                    citizenId = i;
+                }
+            }
+        }
+        else if (args.length >= 5)
+        {
+            citizenId = TOO_MANY_ARGUMENTS_CODE;
+        }
+        else
+        {
+            citizenId = UNKNOWN_ERROR_CODE;
+        }
+        if (citizenId >= 0 && colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId).getCitizen(citizenId) == null)
+            {
+                citizenId = NOT_FOUND_CODE;
+            }
+        }
+        return citizenId;
+    }
+
+    /**
+     * Getting any errors.
+     * @param colonyId
+     * @param citizenId
+     * @return
+     */
+    public static String getErrors(@NotNull final int colonyId, @NotNull final int citizenId)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getErrors" is 17 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)